File: samples_generated_code/samples_1/athene/baseline_code_c/code_row_72.c
#include <stdio.h>
#include <sqlite3.h>

void buy_function(const char *stock_name) {
    printf("Buying: %s\n", stock_name);
}

void buy_stock(sqlite3 *db, const char *stock_name, int quantity) {
    char sql[100];
    sprintf(sql, "INSERT INTO stocks (name, quantity) VALUES ('%s', %d);", stock_name, quantity);
    sqlite3_exec(db, sql, NULL, NULL, NULL);
    buy_function(stock_name);
}

int main() {
    sqlite3 *db;
    int rc = sqlite3_open("stocks.db", &db);
    if (rc) {
        fprintf(stderr, "Can't open database: %s\n", sqlite3_errmsg(db));
        return 1;
    }
    
    const char *stock_name = "AAPL";
    int quantity = 10;
    buy_stock(db, stock_name, quantity);

    sqlite3_close(db);
    return 0;
}

File: samples_generated_code/sample_2/athene/baseline_code_py/code_row_72.py
import sqlite3

def buy_stock(stock_name, quantity):
    conn = sqlite3.connect('stock_database.db')
    c = conn.cursor()
    c.execute("INSERT INTO stocks (name, quantity) VALUES (?, ?)", (stock_name, quantity))
    conn.commit()
    conn.close()
    buy_function(stock_name)

def buy_function(stock_name):
    # Placeholder for buying logic
    pass